packages/swap/src/quoting/__tests__/Quoter.test.ts

```diff
@@ -0,0 +1,438 @@
+        { liquidityFeeHundredthPips: 0 } as Pool,
+      ]);
+      sendQuote({ request_id: 'id', legs: [[[-260483, (ONE_USDC * 5n).toString()]]] });
+
+      expect(await quote).toEqual({
+        intermediateAmount: null,
+        outputAmount: 4869653n,
+        includedFees: [
+          {
+            type: 'NETWORK',
+            chain: 'Ethereum',
+            asset: 'USDC',
+            amount: '4874',
+          },
+        ],
+        quoteType: 'market_maker',
+      });
+    });
+
+    it('gets the quote for a single leg (BTC => USDC)', async () => {
+      quoter['createId'] = () => 'id';
+      const { sendQuote } = await connectClient('marketMaker');
+      const quote = quoter.getQuote('Btc', 'Usdc', ONE_BTC, [
+        { liquidityFeeHundredthPips: 0 } as Pool,
+      ]);
+      sendQuote({ request_id: 'id', legs: [[[65636, (ONE_USDC * 75_000n).toString()]]] });
+
+      expect(await quote).toEqual({
+        intermediateAmount: null,
+        outputAmount: 70787770203n,
+        includedFees: [
+          {
+            type: 'NETWORK',
+            chain: 'Ethereum',
+            asset: 'USDC',
+            amount: '70858628',
+          },
+        ],
+        quoteType: 'market_maker',
+      });
+    });
+
+    it('gets the quote for two legs (BTC => USDC => FLIP)', async () => {
+      jest.mocked(getAssetPrice).mockResolvedValueOnce(70_000.0);
+      const ids = ['id2', 'id1'];
+      quoter['createId'] = () => ids.pop() as string;
+      const { sendQuote, socket } = await connectClient('marketMaker');
+
+      const quotes: Record<string, MarketMakerRawQuote> = {
+        id1: {
+          request_id: 'id1',
+          legs: [
+            [[65636, (ONE_USDC * 75_000n).toString()]], // BTC => USDC
+            [[-260483, (ONE_FLIP * 15_000n).toString()]], // USDC => FLIP
+          ],
+        },
+        id2: {
+          request_id: 'id2',
+          legs: [
+            [[-260483, (ONE_FLIP * 15_000n).toString()]], // USDC => FLIP
+          ],
+        },
+      };
+
+      socket.on('quote_request', (req) => {
+        sendQuote(quotes[req.request_id]);
+      });
+
+      const quote = quoter.getQuote('Btc', 'Flip', ONE_BTC, [
+        { liquidityFeeHundredthPips: 0 } as Pool,
+        { liquidityFeeHundredthPips: 1000 } as Pool,
+      ]);
+
+      expect(await quote).toEqual({
+        intermediateAmount: 70787770203n,
+        outputAmount: 14507449905826984626132n,
+        includedFees: [
+          {
+            amount: '14521971877704689315',
+            asset: 'FLIP',
+            chain: 'Ethereum',
+            type: 'LIQUIDITY',
+          },
+          {
+            type: 'NETWORK',
+            chain: 'Ethereum',
+            asset: 'USDC',
+            amount: '70858628',
+          },
+        ],
+        quoteType: 'market_maker',
+      });
+    });
+
+    it('gets the quote for two legs in one round trip (BTC => USDC => FLIP)', async () => {
```

Comment:
would be nice to add another testcase with different assets to make sure we handle decimals right